### PR TITLE
[SPARK-16133][ML] model loading backward compatibility for ml.feature

### PR DESCRIPTION
**Pull request by hhbyyh**
## What changes were proposed in this pull request?

model loading backward compatibility for ml.feature,
## How was this patch tested?

existing ut and manual test for loading 1.6 models.

### CODE REVIEW

**Comment by mengxr** [comment]
@hhbyyh Since we don't have the test framework to check backward compatibility. Could you confirm that you manually tested loading models saved in 1.6?

File: mllib/src/main/scala/org/apache/spark/ml/feature/MinMaxScaler.scala [hunk 1]
```diff
@@ -232,9 +233,9 @@ object MinMaxScalerModel extends MLReadable[MinMaxScalerModel] {
     override def load(path: String): MinMaxScalerModel = {
       val metadata = DefaultParamsReader.loadMetadata(path, sc, className)
       val dataPath = new Path(path, "data").toString
-      val Row(originalMin: Vector, originalMax: Vector) = sparkSession.read.parquet(dataPath)
-        .select("originalMin", "originalMax")
-        .head()
+      val data = sparkSession.read.parquet(dataPath).select("originalMin", "originalMax")
+      val Row(originalMin: Vector, originalMax: Vector) =
+        MLUtils.convertVectorColumnsToML(data, "originalMin", "originalMax").head()
```
Comment (mengxr) [inline mllib/src/main/scala/org/apache/spark/ml/feature/MinMaxScaler.scala]: Same as I mentioned in #13841, let's convert the columns first and then call select.

**Comment by hhbyyh** [comment]
Thanks for the review. I've manually verified loading 1.6 models.
Updated code and the PR description. Revert change for MaxAbsScaler since it's new in 2.0.

File: mllib/src/main/scala/org/apache/spark/ml/feature/MinMaxScaler.scala [hunk 2]
```diff
@@ -232,7 +233,9 @@ object MinMaxScalerModel extends MLReadable[MinMaxScalerModel] {
     override def load(path: String): MinMaxScalerModel = {
       val metadata = DefaultParamsReader.loadMetadata(path, sc, className)
       val dataPath = new Path(path, "data").toString
-      val Row(originalMin: Vector, originalMax: Vector) = sparkSession.read.parquet(dataPath)
+      val data = sparkSession.read.parquet(dataPath)
+      val Row(originalMin: Vector, originalMax: Vector) = MLUtils.convertVectorColumnsToML(
+        data, "originalMin", "originalMax")
```
Comment (mengxr) [inline mllib/src/main/scala/org/apache/spark/ml/feature/MinMaxScaler.scala]: nit:

~~~scala
val Row(originalMin: Vector, originalMax: Vector) =
  MLUtils.convertVectorColumnsToML(data, "originalMin", "originalMax"))
    .select("originalMin", "originalMax")
~~~

It is slightly hard to read with `data` and `.select` having the same indentation.

Comment (mengxr) [inline mllib/src/main/scala/org/apache/spark/ml/feature/MinMaxScaler.scala]: @hhbyyh Could you fix this style issue?

Comment (hhbyyh) [inline mllib/src/main/scala/org/apache/spark/ml/feature/MinMaxScaler.scala]: Sorry to miss it. Will update right now.

**Comment by hhbyyh** [comment]
@mengxr Thanks for your review. Sent update for the style issue.

**Comment by mengxr** [comment]
LGTM. Merged into master and branch-2.0. Thanks!

**Pull request closed without merging**